src/test/java/org/folio/rest/api/BoundWithStorageTest.java

```diff
@@ -0,0 +1,139 @@
+    IndividualResource aThirdInstance = createInstance("Third Instance");
+    IndividualResource aThirdHoldingsRecord = createHoldingsRecord(aThirdInstance.getId());
+
+    // Make 'item' a bound-with
+    boundWithPartsClient.create(createBoundWithPartJson(mainHoldingsRecord.getId(),item.getId()));
+    IndividualResource secondPart = boundWithPartsClient.create(createBoundWithPartJson(anotherHoldingsRecord.getId(),item.getId()));
+    boundWithPartsClient.create(createBoundWithPartJson(aThirdHoldingsRecord.getId(),item.getId()));
+
+    Response boundWithGETResponseForPartById = boundWithPartsClient.getById(secondPart.getId());
+
+    List<JsonObject> getAllPartsForBoundWithItem = boundWithPartsClient.getByQuery("?query=itemId==" + item.getId());
+
+    assertThat(boundWithGETResponseForPartById.getStatusCode(), is(HttpURLConnection.HTTP_OK));
+    assertThat(getAllPartsForBoundWithItem.size(), is(3));
+  }
+
+  @Test
+  public void cannotDeleteItemThatHasBoundWithParts() {
+    IndividualResource instance1 = createInstance("Instance 1");
+    IndividualResource holdingsRecord1 = createHoldingsRecord(instance1.getId());
+    IndividualResource instance2 = createInstance("Instance 2");
+    IndividualResource holdingsRecord2 = createHoldingsRecord(instance2.getId());
+    IndividualResource item = createItem(holdingsRecord1.getId());
+    boundWithPartsClient.create(createBoundWithPartJson(holdingsRecord1.getId(),item.getId()));
+    boundWithPartsClient.create(createBoundWithPartJson(holdingsRecord2.getId(),item.getId()));
+
+    Response deleteBoundWithItemResponse = itemsClient.attemptToDelete(item.getId());
+    assertThat(deleteBoundWithItemResponse.getStatusCode(), is(HttpURLConnection.HTTP_BAD_REQUEST));
+  }
+
+  @Test
+  public void canChangeOnePartOfABoundWith() {
+    IndividualResource instance1 = createInstance("Instance 1");
+    IndividualResource holdingsRecord1 = createHoldingsRecord(instance1.getId());
+    IndividualResource instance2 = createInstance("Instance 2");
+    IndividualResource holdingsRecord2 = createHoldingsRecord(instance2.getId());
+    IndividualResource item = createItem(holdingsRecord1.getId());
+    boundWithPartsClient.create(createBoundWithPartJson(holdingsRecord1.getId(),item.getId()));
+    IndividualResource part2Created = boundWithPartsClient.create(createBoundWithPartJson(holdingsRecord2.getId(),item.getId()));
+
+    List<JsonObject> getAllPartsForBoundWithItem = boundWithPartsClient.getByQuery("?query=itemId==" + item.getId());
+    List<JsonObject> part2 = boundWithPartsClient.getByQuery("?query=holdingsRecordId==" + holdingsRecord2.getId()+"");
+
+    IndividualResource instance3 = createInstance("Instance 3");
+    IndividualResource holdingsRecord3 = createHoldingsRecord(instance3.getId());
+
+    Response updateResponse = boundWithPartsClient.attemptToReplace(
+      part2Created.getId(),
+      createBoundWithPartJson(holdingsRecord3.getId(), item.getId()));
```

Comment:
What would happen if this changed the record to refer to the same holdings record as the first part?

Is it allowed that a bound-with can have multiple parts that refer to the same holdings record?

Reply:
That would be duplicate entries, since, besides the primary key UUID, there are only two columns in the table, `itemId`and `holdingsRecordId`.  

So it would basically state twice that this title is part of this bound-with. That would not be disastrous perhaps, but messy. If we created a unique index on the two columns, the database would prevent it from happening.

Reply:
> That would not be disastrous perhaps, but messy. If we created a unique index on the two columns, the database would prevent it from happening.

Is that something you are considering as part of this work? If not, are we expecting all clients will accommodate these duplicates?

Reply:
I don't think clients need to accommodate the duplicates. They are just redundant. If they were created manually in the admin, I reckon the librarian would soon discover the typo and delete the duplicate.

Reply:
> Is that something you are considering as part of this work? If not, are we expecting all clients will accommodate these duplicates?

Reply:
> I don't think clients need to accommodate the duplicates. They are just redundant. If they were created manually in the admin, I reckon the librarian would soon discover the typo and delete the duplicate.

We are expecting the people using the system to correct these kinds of issues?

I thought it was mentioned in the meeting we had that the first version of this functionality would not allow editing via the front-end. If that is the case, how would those folks do this?

> If RMB supported composite unique indexes I would just add it but last time I checked I couldn't find such support. We could add an SQL snippet for it but personally I don't feel it's worth the added complexity.

Something like [this](https://github.com/folio-org/mod-circulation-storage/blob/667443265f2a5d427bc8a8bc14dbd2b2f920aeaa/src/main/resources/templates/db_scripts/schema.json#L144-L148) perhaps?

Reply:
Thank you, that looks like it would do what I'd want.

Reply:
> I thought it was mentioned in the meeting we had that the first version of this functionality would not allow editing via the front-end. If that is the case, how would those folks do this?

Without the editing option folks will not create any bound-withs and thus not attach the same title twice. 

The first client populating this structure will likely be GBV and mod-inventory-update.

Reply:
Unique index done.